File: scripts/process_backend/_02/pb02/python/read_file.py
import csv
import time
import sys

filename = sys.argv[1] if len(sys.argv) > 1 else "sample.csv"

start = time.time()

with open(filename, newline='') as f:
    reader = csv.reader(f)
    rows = list(reader)

end = time.time()

print(f"✅ Python read {len(rows)} rows in {end - start:.4f} seconds")

File: scripts/process_backend/_02/pb02/node/read_file.js
const fs = require('fs');
const readline = require('readline');

const filename = process.argv[2] || 'sample.csv';
const lines = [];

const start = process.hrtime();

const rl = readline.createInterface({
    input: fs.createReadStream(filename),
    crlfDelay: Infinity
});

rl.on('line', line => {
    lines.push(line);
});

rl.on('close', () => {
    const [s, ns] = process.hrtime(start);
    const time = s + ns / 1e9;
    console.log(`✅ NodeJS read ${lines.length} rows in ${time.toFixed(4)} seconds`);
});